dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpSessionStore.java

```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.mcp.internal;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import jakarta.enterprise.context.ApplicationScoped;
+
+/**
+ *
+ */
+
+@ApplicationScoped
+public class McpSessionStore {
+
+    private final ConcurrentMap<String, SessionInfo> sessions = new ConcurrentHashMap<>();
+    private static final Duration SESSION_TIMEOUT = Duration.ofMinutes(10);
+
+    public String createSession() {
+        String sessionId = UUID.randomUUID().toString();
+        sessions.put(sessionId, new SessionInfo());
+        return sessionId;
+    }
+
+    public SessionInfo getSession(String sessionId) {
+        return sessions.get(sessionId);
+    }
+
+    public boolean isValid(String sessionId) {
+        cleanupOldSessions();
+        return sessionId != null && sessions.containsKey(sessionId);
+    }
+
+    public void deleteSession(String sessionId) {
+        sessions.remove(sessionId);
+    }
+
+    public void cleanupOldSessions() {
+        Instant now = Instant.now();
+        sessions.entrySet().removeIf(entry -> Duration.between(entry.getValue().getCreatedAt(), now).compareTo(SESSION_TIMEOUT) > 0);
+    }
```

Comment:
This sets a hard cutoff for the lifetime of a session, rather than a time that a session has been idle for. We _probably_ want the latter.